tests/testthat/test-select.R

```diff
@@ -91,3 +91,202 @@ test_that("5D QTables",
     expect_equal(colnames(qtable.5D.flat)[9], "Health-conscious: Diet Coke")
     expect_equal(dim(qtable.5D.flat), dim.target)
 })
+
+
+test_that("Q Tables with vector selection mode",
+{
+    idx <- 1:3
+    out <- SelectFromTable(qtable.2D.xtab,
+                    row.selection.mode = "vector",
+                    row.selections = idx)
+    expect_equivalent(out, qtable.2D.xtab[idx, ])
+    expect_equal(attr(out, "span")$rows,
+                 attr(qtable.2D.xtab, "span")$rows[idx, , drop = FALSE])
+
+    idx <- c(2, 5, 8)
+    out <- SelectFromTable(qtable.2D.xtab,
+                    column.selection.mode = "vector",
+                    column.selections = idx)
+    expect_equivalent(out, qtable.2D.xtab[, idx])
+    expect_equal(attr(out, "span")$columns,
+                 attr(qtable.2D.xtab, "span")$columns[idx, , drop = FALSE])
+
+    ridx <- c("18 to 24", "SUM")
+    expect_warning(out <- SelectFromTable(qtable.2D.xtab,
+                    row.selection.mode = "vector",
+                    row.selections = ridx), "ignored: SUM")
+    expect_equal(nrow(out), 1)
+    expect_equivalent(attr(out, "span")$rows, "18 to 24")
+
+    ridx <- c("Pepsi", "Diet Pepsi", "Pepsi Max")
+    cidx <- c("Older", "Open to new experiences")
+    out <- SelectFromTable(qtable.2D.multi.VS,
+                    row.selection.mode = "vector",
+                    row.selections = ridx,
+                    column.selections = cidx)
+    expect_equal(dim(out), c(length(ridx), length(cidx)))
```

Comment:
Add `expect_equivalent(out, qtable.2D.multi.VS[ridx, cidx])` and also similar for the test above?

Reply:
Done